yarn-project/end-to-end/src/devnet/e2e_devnet.test.ts

```diff
@@ -0,0 +1,226 @@
+
+    const txReceipt = await l2Account
+      .deploy({
+        fee: {
+          gasSettings: GasSettings.default(),
+          paymentMethod: new NativeFeePaymentMethodWithClaim(
+            l2Account.getAddress(),
+            BigInt(claimAmount),
+            Fr.fromBuffer(Buffer.from(claimSecret, 'hex')),
+          ),
+        },
+        skipClassRegistration: false,
+        skipInitialization: false,
+        skipPublicDeployment: false,
+      })
+      .wait();
+
+    expect(txReceipt.status).toBe(TxStatus.SUCCESS);
+    const feeJuice = await GasTokenContract.at(
+      (
+        await pxe.getNodeInfo()
+      ).protocolContractAddresses.gasToken,
+      txReceipt.wallet,
+    );
+    const balance = await feeJuice.methods.balance_of_public(l2Account.getAddress()).simulate();
+    expect(balance).toEqual(amount - txReceipt.transactionFee!);
+  });
+
+  type OptionValue = undefined | boolean | { toString(): string };
+  type ArgumentValue = { toString(): string };
+
+  function cli<T>(cliCommand: string): Promise<T>;
+  function cli<T>(cliCommand: string, args: ArgumentValue[]): Promise<T>;
+  function cli<T>(cliCommand: string, opts: Record<string, OptionValue>): Promise<T>;
+  function cli<T>(cliCommand: string, args: ArgumentValue[], opts: Record<string, OptionValue>): Promise<T>;
+  function cli<T>(
+    cliCommand: string,
+    _args?: ArgumentValue[] | Record<string, OptionValue>,
+    _opts?: Record<string, OptionValue>,
+  ): Promise<T> {
+    const { promise, resolve, reject } = promiseWithResolvers<T>();
+    const args = Array.isArray(_args) ? _args : [];
+    const opts = _args && !Array.isArray(_args) ? _args : typeof _opts !== 'undefined' ? _opts : {};
+
+    const cmdArguments = args.map(x => x.toString());
+
+    opts.json = true;
+    const cmdOptions = Object.entries(opts)
+      .filter((entry): entry is [string, { toString(): string }] => typeof entry[1] !== 'undefined')
+      .map(([key, value]) =>
+        typeof value === 'boolean' ? (value ? `--${key}` : `--no-${key}`) : `--${key} ${value.toString()}`,
+      );
+
+    const cmd = `${AZTEC_CLI} ${cliCommand} ${cmdArguments.join(' ')} ${cmdOptions.join(' ')}`;
+    logger.info(`Running aztec-cli: ${cmd}`);
+    const child = exec(cmd, {
+      cwd: tmpdir(),
+      env: {
+        NODE_OPTIONS: '--no-warnings',
+      },
+    });
+
+    let err = '';
+    child.stderr?.on('data', data => {
+      logger.error(data.toString());
+      err += data.toString();
+    });
+
+    let out = '';
+    child.stdout?.on('data', data => {
+      out += data.toString();
+    });
+
+    child.on('error', reject);
+    child.on('close', (code, signal) => {
+      if (code === 0) {
+        const res = JSON.parse(out);
+        logger.info(`aztec-cli JSON output: ${JSON.stringify(res)}`);
+        resolve(res);
+      } else {
+        reject(new Error(`aztec-cli non-zero exit: code=${code} signal=${signal} ${err}`));
+      }
+    });
```

Comment:
I understand [`execSync`](https://nodejs.org/api/child_process.html#child_processexecsynccommand-options) does most of this stuff for you. No neeed to change though.

Reply:
I had a look at `execSync`, on the success case it only returns `stdout` so we lose all debug logging from the CLI 🥲